test/integration/integration.cc

```diff
@@ -918,6 +918,23 @@ void BaseIntegrationTest::testNoHost() {
   EXPECT_TRUE(response.find("HTTP/1.1 400 Bad Request\r\n") == 0);
 }
 
+void BaseIntegrationTest::testAbsolutePath() {
+  Buffer::OwnedImpl buffer("GET http://www.redirect.com HTTP/1.1\r\nHost: host\r\n\r\n");
+  std::string response;
+  RawConnectionDriver connection(
+                                  lookupPort("http_forward"), buffer,
+                                  [&](Network::ClientConnection& client, const Buffer::Instance& data) -> void {
+                                    response.append(TestUtility::bufferToString(data));
+                                    client.close(Network::ConnectionCloseType::NoFlush);
+                                  },
+                                  version_);
+
+  connection.run();
+  EXPECT_FALSE(response.find("HTTP/1.1 404 Not Found\r\n") == 0);
+}
+
+
+
```

Comment:
I think you'll definitely want CONNECT tests too - I'm not completely sure if envoy will make the correct assumption that all other following payload should be through-proxied as if it were body data.